GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -798,9 +804,32 @@ private bool IsValidRevisionIndex(int index)
             return _gridView.GetRevision(row);
         }
 
-        public GitRevision GetCurrentRevision()
+        /// <summary>
+        /// Get the actual GitRevision from grid or use GitModule if parents may be rewritten or the commit is not in the grid.
+        /// </summary>
+        /// <returns>The GitRevision or null if not found</returns>
+        public GitRevision? GetActualRevision(ObjectId objectId)
```

Comment:
Why is this public?

Reply:
This is used in #9424 (and #9445).

Reply:
As a general rule I prefer enable functionality when it is required. It is impossible to know what and when may be used (or unused) in the future.